### PR TITLE
Support for custom agents

### PR DESCRIPTION
**Pull request by vineetshar**
Screenshot 2024-10-09 at 11 23 42 AM
Screenshot 2024-10-09 at 11 23 47 AM
Screenshot 2024-10-09 at 6 32 47 PM
Screenshot 2024-10-09 at 6 51 47 PM
Screenshot 2024-10-09 at 11 23 42 AM

### CODE REVIEW

File: app/modules/intelligence/agents/agents_schema.py [hunk 1]
```diff
@@ -1,7 +1,35 @@
...
 
 class AgentInfo(BaseModel):
     id: str
     name: str
     description: str
+
+
+class TaskCreate(BaseModel):
+    description: str
+    expected_output: str
```
Comment (dhirenmathur) [inline app/modules/intelligence/agents/agents_schema.py]: Tasks should have optional tools parameter as well

**Review by dhirenmathur — CHANGES_REQUESTED** [review]
LGTM - just want to add tool param to tasks

File: app/modules/intelligence/agents/agents_schema.py (hunk 1 above)
Comment (vineetshar) [reply app/modules/intelligence/agents/agents_schema.py]: fixed

**Review by dhirenmathur — CHANGES_REQUESTED** [review]
One additional req : Lets validate that expected output is a valid json

File: app/modules/intelligence/agents/agents_schema.py [hunk 2]
```diff
@@ -11,8 +11,16 @@ class AgentInfo(BaseModel):
...
     tools: List[str]
+    expected_output: Any = Field(description="A JSON object")
+    tools: List[str]
+
+    @field_validator("expected_output")
+    @classmethod
+    def validate_json_object(cls, v):
+        if not isinstance(v, dict):
+            raise ValueError("Expected a JSON object")
+        return v
```
Comment (dhirenmathur) [inline app/modules/intelligence/agents/agents_schema.py]: Pydantic has no json type?

**Review by dhirenmathur — APPROVED** [review]

**Pull request merged**